### PR TITLE
fixes for occ sql, time partitions, remote sql following code static …

### PR DESCRIPTION
**Pull request by dorinhogea**
…analysis

### CODE REVIEW

File: db/osqlrepository.c [hunk 1]
```diff
@@ -305,12 +305,11 @@ int osql_repository_rem(osql_sess_t *sess, int lock, const char *func, const cha
                     p, found_uuid);
         }
 
-        abort();
         if (lock) {
```
Comment (adizaimi) [inline db/osqlrepository.c]: If we are aborting, why unlock?

File: db/translistener.c [hunk 2]
```diff
@@ -1166,22 +1166,31 @@ int javasp_load_procedure_int(const char *name, const char *param,
...
+    }
+    if (paramvalue) {
         p->qname = strdup(name);
+        if(!p->qname) goto oom;
+    }
     if (param)
         p->param = strdup(param);
     else
         p->param = strdup("<sc>");
+    if(!p->param) goto oom;
```
Comment (adizaimi) [inline db/translistener.c]: Does not really matter, but we are not freeing p->qname if we goto oom from here. Nor p->name either.

**Comment by adizaimi** [comment]
looks good to me.

File: db/osqlrepository.c (hunk 1 above)
Comment (dorinhogea) [inline db/osqlrepository.c]: Looks like the abort is meant more like a temporary patch that will allow future investigation in what produce a certain type of error (and probably we didn't figure out how to unwind properly the error).

File: db/translistener.c (hunk 2 above)
Comment (dorinhogea) [inline db/translistener.c]: gonna fix that anyway

**Comment by riverszhang89** [comment]
1. Please squash less meaningful commits into one.
2. Please merge your branch after passing regression.

**Pull request merged**